linalg/densemat.hpp

```diff
@@ -58,6 +58,20 @@ class DenseMatrix : public Matrix
    DenseMatrix(double *d, int h, int w)
       : Matrix(h, w) { UseExternalData(d, h, w); }
 
+   /// Create a dense matrix using a braced initializer list
+   template <int M, int N>
+   explicit DenseMatrix(const double (&values)[M][N]) : DenseMatrix(M, N)
+   {
+      // DenseMatrix is column-major so copies have to be element-wise
+      for (int i = 0; i < M; i++)
+      {
+         for (int j = 0; j < N; j++)
+         {
+            Elem(i,j) = values[i][j];
```

Comment:
```suggestion
            (*this)(i,j) = values[i][j];
```
Can avoid virtual function call